loader/loader.c

```diff
@@ -2532,6 +2570,10 @@ static VkResult loader_read_layer_json(const struct loader_instance *inst, struc
     if (VK_SUCCESS != result && NULL != props) {
         // Make sure to free anything that was allocated
         loader_remove_layer_in_list(inst, layer_instance_list, props_index);
+        if (layer_instance_list->count == 0) {
+            // Cleanup any allocated memory
+            loader_delete_layer_list_and_properties(inst, layer_instance_list);
+        }
```

Comment:
Why are we removing the *entire* layer list if only one of the entries has a non-success return value? Shouldn't it only do this for catastrophic errors?

I worry about the case where a layer is 'invalid' and it nukes the rest of  the list.

Reply:
I think it was related to my bad code for the above.  Undoing.